doc/component.md

```diff
@@ -6,8 +6,8 @@ class YourComponent extends Component {
   int yourProperty;
 
   @override
-  void init([InitObject data]) {
-    yourPoperty = 10;
+  void init([data]) {
```

Comment:
would `void init([YourInitDataType data]) {` be more clear here?

Reply:
In this example the `Component` has no generic data type, so if you auto complete it, it will also show it likes this

Reply:
but what is the variable type of data? I thought dart would infer dynamic, which is bad

Reply:
It is dynamic, because the extended class does not receive a generic data type. 

If I write this code without the methods, my IDE will automatically generate them without any types because it can't resolve it. There is not much we can do about it because Dart resolves the generic as dynamic automatically.